packages/core/src/dispatcher.ts

```diff
@@ -123,9 +126,13 @@ export default class Dispatcher extends EventEmitter {
       const cid = await this._ipfs.dag.put(jws, { format: 'dag-jose', hashAlg: 'sha2-256' })
       // put the payload into the ipfs dag
       await this._ipfs.block.put(linkedBlock, { cid: jws.link.toString() })
+      await this._restrictRecordSize(jws.link.toString())
+      await this._restrictRecordSize(cid)
```

Comment:
This is an aside and doesn't need to be done now, but figured I'd mention it while I'm thinking of it: longer term I think it'd be nice to put an interface in front of IPFS, like you did in the CAS.  Then the logic for enforcing record size could live there so we don't have to duplicate it every time we put or fetch an object from IPFS

Reply:
Yeah, that makes sense. We could decouple the `Dispatcher` from the IPFS service.

Reply:
@simonovic86 the `cid` in this snippet will never be to large fyi. The payload is always the CID that is represented in the `link` property.

Reply:
yeah, I know. Just wanted to be sure 🙂